src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java

```diff
@@ -0,0 +1,104 @@
+    return false;
+  }
+
+  public static String getPatientId(String json) throws JsonProcessingException {
+    ObjectMapper mapper = new ObjectMapper();
+
+    JsonNode jsonNode = mapper.readTree(json);
+    JsonNode entry = jsonNode.get("entry");
+    JsonNode theNode = null;
+    Iterator<JsonNode> entyIter = entry.iterator();
+    String existingPatientId = null;
+    while (entyIter.hasNext()) {
+      theNode = entyIter.next();
+      var resourceNode = theNode.get("resource");
+      if (resourceNode != null) {
+        var resourceType = resourceNode.get("resourceType");
+        if (resourceType != null && "PATIENT".equalsIgnoreCase(resourceType.asText())) {
+          existingPatientId = resourceNode.get("id").textValue();
+        }
+      }
+    }
+    return existingPatientId;
+  }
+
+  public static String getPatientFullUrl(String json) throws JsonProcessingException {
+    ObjectMapper mapper = new ObjectMapper();
+
+    JsonNode jsonNode = mapper.readTree(json);
+    JsonNode entry = jsonNode.get("entry");
+    JsonNode theNode = null;
+    Iterator<JsonNode> entyIter = entry.iterator();
+    String fullUrl = null;
+    while (entyIter.hasNext()) {
+      theNode = entyIter.next();
+      var resourceNode = theNode.get("resource");
+      if (resourceNode != null) {
+        var resourceType = resourceNode.get("resourceType");
+        if (resourceType != null
+            && "PATIENT".equalsIgnoreCase(resourceType.asText())
+            && theNode.has("fullUrl")) {
+          fullUrl = theNode.get("fullUrl").asText();
+        }
+      }
+    }
+    return fullUrl;
+  }
+
+  public static String updateFullUrl(
+      final String fullUrl, final String oldPatientId, final String newPatientId) {
+    if (!StringUtils.isBlank(fullUrl) && fullUrl.endsWith(oldPatientId)) {
+      return fullUrl.substring(0, fullUrl.length() - oldPatientId.length()) + newPatientId;
+    }
+    return fullUrl;
+  }
+
+  public static String replacePatientRefs(String json, String oldPatientId, String newPatientId) {
+    final String pattern = "(?i)(\"reference\"\\s*:\\s*\"Patient/" + oldPatientId + "\")";
+    return json.replaceAll(pattern, "\"reference\": \"Patient/" + newPatientId + "\"");
+  }
+
+  public static String replacePatientRefs(String json, String newPatientId) {
+    final String pattern = "(?i)(\"reference\"\\s*:\\s*\"Patient\\/[A-Za-z0-9\\-\\.]{1,64}\")";
+    return json.replaceAll(pattern, "\"reference\": \"Patient/" + newPatientId + "\"");
+  }
+
+  public static String replaceFullUrlRefs(String json, String oldFullUrl, String newPatientId) {
+    final String pattern = "(\"reference\"\\s*:\\s*\"" + Pattern.quote(oldFullUrl) + "\")";
+    return json.replaceAll(pattern, "\"reference\": \"Patient/" + newPatientId + "\"");
+  }
+
+  public static boolean isUuid(String value) {
+    try {
+      UUID.fromString(value);
+      return true;
+    } catch (Exception exception) {
+      // handle the case where string is not valid UUID
```

Comment:
Do we have to handle this ?

Reply:
no...lol bad comment - in this case we don't care, we just return false to indicate that it is not a UUID